src/ringct/rctSigs.cpp

```diff
@@ -1578,6 +1579,49 @@ namespace rct {
       }
     }
 
+    bool verRctNonSemanticsSimpleCached(const rctSig & rv)
+    {
+      // Hello future Monero dev! If you got this assert, read the following carefully:
+      //
+      // RCT cache assumes that this function will serialize and hash all rv's fields used for RingCT verification
+      // If you're about to add a new RCTType here, first you must check that binary_archive serialization writes all rv's fields to the binary blob
+      // If it's not the case, rewrite this function to serialize everything, even some "temporary" fields which are not serialized normally
+      CHECK_AND_ASSERT_MES_L1(rv.type <= RCTTypeBulletproofPlus, false, "Unknown RCT type. Make sure RCT cache works correctly with this type and then enable it in the code here.");
+
+      // Don't cache older (or newer) rctSig types
+      // This cache only makes sense when it caches data from mempool first,
+      // so only "current fork version-enabled" RCT types need to be cached
+      if (rv.type != RCTTypeBulletproofPlus)
+        return verRctNonSemanticsSimple(rv);
+
+      // Get the hash of rv
+      std::stringstream ss;
+      binary_archive<true> ar(ss);
+
+      rctSig& d = const_cast<rctSig&>(rv);
+
+      // Explicitly serialize "not serialized, it can be reconstructed" fields here
+      ::do_serialize(ar, d.message);
```

Comment:
There's two serialization functions, [the other](https://github.com/monero-project/monero/blob/master/src/ringct/rctTypes.h#L388) is used by `do_serialize`. So both of these fields should be included in that serialization afaik. Normally the other one is called though (again, afaik).

Reply:
I think you're right, I need to double check this. `serialize_rctsig_base` is only called for serializing transactions, not through `do_serialize`. Then these 2 lines are not necessary.

Reply:
It serializes whatever is listed inside `BEGIN_SERIALIZE/BEGIN_SERIALIZE_OBJECT`, so these 2 lines are not needed.

Reply:
For reference: here is where `message` and `mixRing` are serialized through `do_serialize()` to `rctSig`:

https://github.com/monero-project/monero/blob/9367b432f69869dd02ca4f0b1feceb2674e4cd3b/src/ringct/rctTypes.h#L390-L391

https://github.com/monero-project/monero/blob/9367b432f69869dd02ca4f0b1feceb2674e4cd3b/src/ringct/rctTypes.h#L609